marketplace-and-integrations/packages/authorized-services.md

```diff
@@ -0,0 +1,776 @@
+
+###### AccessTokenResponseKey
+
+The expected key for retrieving an access token from a response. If not provided the default `access_token` is assumed.
+
+###### RefreshTokenResponseKey
+
+The expected key for retrieving a refresh token from a response. If not provided the default `refresh_token` is assumed.
+
+###### ExpiresInResponseKey
+
+The expected key for retrieving the datetime of token expiry from a response. If not provided the default `expires_in` is assumed.
+
+###### SampleRequest
+
+An optional sample request can be provided, which can be used to check that an authorized service is functioning as expected from the backoffice.  For example, to retrieve the set of contributors to the Umbraco repository hosted at GitHub, this request can be used: `/repos/Umbraco/Umbraco-CMS/contributors`.
+
+### Authorizing a Service
+
+With one or more service configured, it will be available from the items within a tree in the _Settings_ section.
+
+Clicking on an item will show some details about the configured service, and it's authentication status.
+
+If the service is not yet authorized, click the _Authorize Service_ button to trigger the authentication and authorization flow. You will be directed to the service to login, and optionally choose an account.  You will then be asked to agree to the permissions requested by the app. Finally you will be redirected back to the Umbraco backoffice. You should see confirmation that an access token has been retrieved and stored such that the service is now authorized. If provided, you can click the _Verify Sample Request_ button to ensure that service's API can be called.
+
+### Calling an Service
+
+To make a call to an authorized service, you first need to obtain an instance of `IAuthorizedServiceCaller`. This is registered with the dependency injection framework. As such it can be injected into a controller, view or service class where it needs to be used.
+
+If making a request where all information is provided via the path and querystring, such as GET requests, the following method should be invoked:
+
+```csharp
+Task<TResponse> SendRequestAsync<TResponse>(string serviceAlias, string path, HttpMethod httpMethod);
+```
+
+The parameters for the request are as follows:
+
+- `serviceAlias` - the alias of the service being invoked (e.g. `github`).
+- `path` - the path to the API method being invoked (e.g. `/repos/Umbraco/Umbraco-CMS/contributors`).
+- `httpMethod` - the HTTP method to use for the request (e.g. `HttpMethod.Get`).
+
+There is also a type parameter:
+- `TResponse` - defines the strongly typed representation of the service method's response, that the raw response content will be deserialized into.
+
+If you need to provide data in the request an overload is available. This can be used for POST or PUT requests that trigger the creation or update of a resource:
```

Comment:
```suggestion
If you need to provide data in the request an overload is available. This can be used for `POST` or `PUT` requests that trigger the creation or update of a resource:
```